managed/models/scheduled_tasks_helpers.go

```diff
@@ -266,3 +296,31 @@ func checkUniqueScheduledTaskID(q *reform.Querier, id string) error {
 		return errors.WithStack(err)
 	}
 }
+
+func nameFromTaskData(taskType ScheduledTaskType, taskData *ScheduledTaskData) (string, error) {
+	if taskData != nil {
+		switch taskType {
+		case ScheduledMySQLBackupTask:
+			if taskData.MySQLBackupTask != nil {
+				return taskData.MySQLBackupTask.Name, nil
+			}
+		case ScheduledMongoDBBackupTask:
+			if taskData.MongoDBBackupTask != nil {
+				return taskData.MongoDBBackupTask.Name, nil
+			}
+		default:
+			return "", status.Errorf(codes.InvalidArgument, "Unknown type: %s", taskType)
+		}
+	}
+	return "", nil
+}
+
+func isNameAlreadyExistsErr(err error, name string) bool {
+	if err, ok := err.(*pq.Error); ok {
+		// We check like this because there is no structured info in the returned error.
+		if err.Code == pgUniqueConstraintViolationCode && strings.Contains(err.Detail, name) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I guess because of that we have `checkUnique*` functions. While this may work too(i'm not sure about "contains" part), isn't it better to use same approach as everywhere in code base?